src/ImageData/FileLoader.cc

```diff
@@ -870,6 +875,15 @@ double FileLoader::CalculateBeamArea() {
     return info.getBeamAreaInPixels(-1, -1, _coord_sys->directionCoordinate());
 }
 
+void FileLoader::NormalizeBunit() {
+    if (_image) {
+        auto bunit = _image->units().getName();
+        bunit.erase(std::remove(bunit.begin(), bunit.end(), '"'), bunit.end());
+        NormalizeUnit(bunit);
+        _image->setUnits(bunit);
+    }
+}
+
```

Comment:
Now that you are only calling this from one place (the base file loader's `OpenFile`) I don't think it's necessary to split this into a function anymore. I suggest that you put this code directly in `OpenFile`.

Did you re-test all the capitalization cases? If I understand @pford's comments correctly, the `NormalizeUnit` function won't correct all of them, so some custom handling may be required.

Overall I think that this is OK now, but I will also wait for @pford to have a look.

Reply:
Unless I missed something, `NormalizeUnit` will fix all cases.  The input string is converted to all caps (*before* checking), then checked (all valid units will pass as all caps), then converted `fromFITS` to the proper mixed- or lower-case.

Reply:
Oh, good. Then this should be fine. I'll ask @ajm to re-test to verify, and then we can merge this.